src/librustc_driver/lib.rs

```diff
@@ -356,27 +357,66 @@ fn handle_explain(code: &str,
     match descriptions.find_description(&normalised) {
         Some(ref description) => {
             let mut is_in_code_block = false;
+            let mut text = String::new();
+
             // Slice off the leading newline and print.
             for line in description[1..].lines() {
                 let indent_level = line.find(|c: char| !c.is_whitespace())
                     .unwrap_or_else(|| line.len());
                 let dedented_line = &line[indent_level..];
                 if dedented_line.starts_with("```") {
                     is_in_code_block = !is_in_code_block;
-                    println!("{}", &line[..(indent_level+3)]);
+                    text.push_str(&line[..(indent_level+3)]);
+                    text.push('\n');
```

Comment:
IMO It's ok to do that `push()`. Though you are doing it here and in like `375`, so you are adding a carriage return regardless of the `if` statement, I would just put a single one between lines `376`-`377` instead, but that is totally a **nitpick**.